objectgen/objectgen/object_def.py

```diff
@@ -15,6 +15,7 @@ class ObjectField:
     field_name: str
     field_type: Type
     is_binding: bool = False
+    check_equality: bool = True
```

Comment:
Would it make sense for this to be called `is_metadata`?

Reply:
In the case of Spans, that makes sense, but I also had to set this false for function names. if function names can also count as metadata I think it works

Reply:
arguably function names are meta data if you are only looking at structure

Reply:
I think we should keep it generic potentially like use_in_{equal/hash/etc} this is pretty much deriving for all of object gen so it shouldn't be specific to AST imo